sql-db/narayana-transactions/src/main/java/io.quarkus.ts.transactions/TransferTransactionService.java

```diff
@@ -0,0 +1,52 @@
+package io.quarkus.ts.transactions;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+import org.jboss.logging.Logger;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.narayana.jta.QuarkusTransaction;
+
+@ApplicationScoped
+@Named("regular-transaction")
+public class TransferTransactionService extends TransferProcessor {
+
+    private static final Logger LOG = Logger.getLogger(TransferTransactionService.class);
+    private final static String ANNOTATION_TRANSACTION = "user transaction to other user";
+    private final static int TRANSACTION_TIMEOUT_SEC = 10;
+    private final MeterRegistry registry;
+    private long transactionsAmount;
+
+    public TransferTransactionService(MeterRegistry registry) {
+        this.registry = registry;
+        registry.gauge("transaction.regular.amount", this, TransferTransactionService::getTransactionsAmount);
+    }
+
+    public JournalEntity makeTransaction(String from, String to, int amount) {
+        JournalEntity journal = null;
+        LOG.infof("Regular transaction, from %s to %s amount %s", from, to, amount);
+        try {
+            // please don't move this gauge after commit statement, because we want to test the gauges after a rollback
+            transactionsAmount++;
+            verifyAccounts(from, to);
+            QuarkusTransaction.begin(QuarkusTransaction.beginOptions().timeout(TRANSACTION_TIMEOUT_SEC));
+            accountService.decreaseBalance(from, amount);
+            accountService.increaseBalance(to, amount);
+            journal = journalService.addToJournal(from, to, ANNOTATION_TRANSACTION, amount);
```

Comment:
I suggest to call addToJournal before making operations on the account in all services.
Right now, all tests expect line decreaseBalance to fail. In that case, increase of balance and addition to journal do happen, and therefore are not rolled back. I presume, that if you comment out all transaction code in the services, tests will pass anyway.
If me will write to journal before failure, then we would check this rollback for real.